org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/UiSession.java

```diff
@@ -605,6 +611,52 @@ public void dispose() {
     m_sessionMetrics.sessionDestroyed(SESSION_TYPE);
   }
 
+  protected void runInModelJob(Runnable r, String name) {
+    runInModelJob(r, name, false);
+  }
+
+  protected void runInModelJob(Runnable r, String name, boolean wait) {
+    if (r == null) {
+      return;
+    }
+
+    // current thread is the model thread -> run
+    if (ModelJobs.isModelThread()) {
+      r.run();
+      return;
+    }
+
+    // current thread is NOT the model thread and the client session is active -> run in model job
+    final IClientSession clientSession = getClientSession();
+    if (clientSession != null && clientSession.isActive() && !clientSession.isStopping()) {
+      final ClientRunContext clientRunContext = ClientRunContexts.copyCurrent(true).withSession(clientSession, true);
+      IFuture<Void> modelJob = ModelJobs.schedule(
+          r::run,
+          ModelJobs.newInput(clientRunContext)
+              .withName(name)
+              .withExceptionHandling(null, false));// Propagate exception to caller (UIServlet)
+
+      // no need to wait -> return
+      if (!wait) {
+        return;
+      }
+
+      try {
+        // wait for the model job to complete
+        modelJob.awaitDone(2, TimeUnit.SECONDS);
+        return;
+      }
+      catch (TimedOutError e) {
+        // the model job did not complete within wait time -> cancel it and do NOT return but run task in current thread
+        modelJob.cancel(true);
+        LOG.debug("Unable to run task \"{}\" in model job.", name);
```

Comment:
- Move log line to start of catch block
- Shouldn't the log level be at least INFO? This should happen very rarely and it would probably be good to see if it doesn't.
- We usually write ' inside log strings to avoid the escape characters